src/morphology/english/getForms.js

```diff
@@ -1,11 +1,13 @@
-const getNounForms = require( "./getNounForms.js" ).getNounForms;
+const getNounFormsWithPossessives = require( "./getNounForms.js" ).getNounFormsWithPossessives;
+const checkPossessive = require( "./getNounForms.js" ).checkPossessive;
 const getVerbForms = require( "./getVerbForms.js" ).getVerbForms;
 const unique = require( "lodash/uniq" );
 
 const getForms = function( word ) {
-	let forms = [].concat( getNounForms( word ), getVerbForms( word ) );
-
-	return unique( forms );
+	if ( checkPossessive( word ) ) {
+		return unique( getNounFormsWithPossessives( word ) );
```

Comment:
`checkPossessive` and `getNounFormsWithPossessive` are running the same regex (`possessiveToBaseRegex`). So basically they're doing the same job, except that one returns a boolean and the other also carries out replacement. Would it be possible to change it so that `getNounFormsWithPossessive` does the checking and replaces if applicable, and does nothing if it's not applicable?  Now the same regex is running twice, which seems a bit redundant.

Reply:
I removed the redundancy in terms of calling the same regex twice. However, I kept the function `checkPossessive` because it is used in another function of this script (`getNounFormsWithPossesives`) and outside the script (in `getForms.js`).